### DISCUSSION TITLE
Proposal: Megarepo-ify annexes, just like packages

### DISCUSSION THREAD

**Discussion by alichtman** [zdharma-continuum/zinit#92] [Ideas]
As title.

**Comment by pschmitt** [comment]
Interesting idea, I totally get the appeal.

The downside of a monorepo would be that the ices required to fetch and use the annexes will need to be updated. That can easily be fixed if we provide dedicated (a) package(s).

I'm not so sure about the outcome:

- Would `zinit load zdhc/zinit-annexes` load *all* annexes? Given what the man annex does we probably don't want to do that.

**Reply by alichtman** [reply]
>  downside of a monorepo would be that the ices required to fetch and use the annexes will need to be updated

Yes, but I think the benefits in terms of organization and discoverability are worth it. We would need to change the syntax slightly (and the backend changes that correspond with it): 

```
zinit load zdhc/zinit-annexes/rust
```

And `zinit load zdhc/zinit-annexes` could load all annexes, sure.

**Reply by pschmitt** [reply]
We don't **need** to support `ORG/REPO/DIR` even though I like the idea (can you open an issue with that?).

`zinit load pick"rust/rust.plugin.zsh" zdhc/zinit-annexes` should work fine already :)

**Reply by alichtman** [reply]
> We don't need to support ORG/REPO/DIR even though I like the idea

I do want to be careful about not providing *too many* ways to do something.

I guess I'm ok with supporting both of:

```
zinit load pick"rust/rust.plugin.zsh" zdhc/zinit-annexes
zinit load pick"rust.plugin.zsh" zdhc/zinit-annexes/rust
```

but I like the pythonic / Go approach of "there is one, and only one, right way to do X operation"
